js/sign/src/signers/integrity-block-signer.ts

```diff
@@ -16,28 +17,22 @@ type IntegritySignature = {
   signature: Uint8Array;
 };
 
-type IntegrityBlockSignerOptions = {
-  key: KeyObject;
-};
-
 export class IntegrityBlockSigner {
-  private key: KeyObject;
   private webBundle: Uint8Array;
+  private signingStrategy: ISigningStrategy;
 
-  constructor(webBundle: Uint8Array, opts: IntegrityBlockSignerOptions) {
-    if (opts.key.asymmetricKeyType !== 'ed25519') {
-      throw new Error(
-        `IntegrityBlockSigner: Only ed25519 keys are currently supported. Your key's type is ${opts.key.asymmetricKeyType}.`
-      );
-    }
-    this.key = opts.key;
+  constructor(webBundle: Uint8Array, signingStrategy: ISigningStrategy) {
```

Comment:
nit: Use `private readonly` for both parameters and remove the private members explicitly declared above and the assignments in the constructor body.